core/domain/prod_validation_jobs_one_off.py

```diff
@@ -288,6 +321,8 @@ def _validate_related_model_properties(cls, item):
 
             for (_, _, related_model) in (
                     related_model_class_model_id_model_tuples):
+                if related_model is None or related_model.deleted:
```

Comment:
Er ... this seems wrong? At the very least it needs explanation.

Reply:
When we are fetching models, we get all the models and there might be None instances or deleted instances. So, this line is to be avoid checking properties of such instances. This is the same check as we do while testing if models corresponding to external ids are missing [here](https://github.com/oppia/oppia/pull/6995/files/dde0ffc5a391e97b4b53fa8c21ff7c675a1209a5#diff-ecafbbaeb117bcae187e78e5b33e120fR202). I will add a comment to explain this.

Reply:
But I don't get it ... if a related model is missing, or a model corresponding to an external ID is missing, shouldn't that count as an error and be handled accordingly? E.g. we had a problem like that recently for activity references containing references to deleted explorations.

Reply:
Yes, but that is already being done [here](https://github.com/oppia/oppia/pull/6995/files/dde0ffc5a391e97b4b53fa8c21ff7c675a1209a5#diff-ecafbbaeb117bcae187e78e5b33e120fR191).

The test which is happening in these lines is to validate if the properties in a summary model match the corresponding properties in a base model.

Reply:
These changes were merged in the first PR but I just thought of adding this while testing a deleted model instance in test.

Reply:
Yep, I still have your first PR on my plate to review fully :)

Could you respond more fully to my question here? https://github.com/oppia/oppia/pull/6995/files/dde0ffc5a391e97b4b53fa8c21ff7c675a1209a5#r296470915 I think this is a conceptual point that we should ensure we have the same understanding on.

Reply:
I am sorry I did not get your doubt. Is something unclear in the reply [here](https://github.com/oppia/oppia/pull/6995#discussion_r296470983)?

I can give an example here to be more clear: If we have a **model1** which requires an external **model2** but let us say the **model2** is deleted. 

Now, the check to find that **model2** is deleted is done by the function [_validate_external_id_relationships](https://github.com/oppia/oppia/pull/6995/files/dde0ffc5a391e97b4b53fa8c21ff7c675a1209a5#diff-ecafbbaeb117bcae187e78e5b33e120fR191). 

Also, this **model2** will be fetched in `external_instance_details` but it's deleted field will be true. In all other tests where we are using the model2 to validate a related property, I have just added a check to see if **model2** is not None or deleted, as in that case there is no use of validating the related property.

Reply:
Hi @ankita240796 per our discussion in the call, let's do the renaming of "related" to "external" to reduce confusion, and add comments as needed. Thanks!

Reply:
Yes, I have updated that but I have done it in the common PR #6989 instead since the change was required for all models (adding comments everywhere) and stacked this on that PR.